src/ImmutableX.ts

```diff
@@ -0,0 +1,842 @@
+    return this.workflows.deposit(ethSigner, deposit).catch(err => {
+      throw formatError(err);
+    });
+  }
+
+  /**
+   * Get details of a Deposit with the given ID
+   * @param request - the request object containing the parameters to be provided in the API request
+   * @returns a promise that resolves with the requested Deposit
+   * @throws {@link index.IMXError}
+   */
+  public getDeposit(request: DepositsApiGetDepositRequest) {
+    return this.depositsApi
+      .getDeposit(request)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Get a list of Deposits
+   * @param request - the request object containing the parameters to be provided in the API request
+   * @returns a promise that resolves with the requested list of Deposits
+   * @throws {@link index.IMXError}
+   */
+  public listDeposits(request?: DepositsApiListDepositsRequest) {
+    return this.depositsApi
+      .listDeposits(request)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Register a User to Immutable X if they are not already
+   * @param walletConnection - the pair of L1/L2 signers
+   * @returns a promise that resolves with void if successful
+   * @throws {@link index.IMXError}
+   */
+  public registerOffchain(walletConnection: WalletConnection) {
+    return this.workflows.registerOffchain(walletConnection).catch(err => {
+      throw formatError(err);
+    });
+  }
+
+  /**
+   * Checks if a User is registered on on-chain
+   * @param walletConnection - the pair of L1/L2 signers
+   * @returns a promise that resolves with true if the User is registered, false otherwise
+   * @throws {@link index.IMXError}
+   */
+  public isRegisteredOnchain(walletConnection: WalletConnection) {
+    return this.workflows.isRegisteredOnchain(walletConnection).catch(err => {
+      throw formatError(err);
+    });
+  }
+
+  /**
+   * Get Stark keys for a registered User
+   * @param ethAddress - the eth address of the User
+   * @returns a promise that resolves with the requested User
+   * @throws {@link index.IMXError}
+   */
+  public getUser(ethAddress: string) {
+    return this.usersApi
+      .getUsers({ user: ethAddress })
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Get details of an Asset
+   * @param request - the request object containing the parameters to be provided in the API request
+   * @returns a promise that resolves with the requested Asset
+   * @throws {@link index.IMXError}
+   */
+  public getAsset(request: AssetsApiGetAssetRequest) {
+    return this.assetApi
+      .getAsset(request)
+      .then(res => res.data)
+      .catch(err => {
+        throw formatError(err);
+      });
+  }
+
+  /**
+   * Get a list of Assets
+   * @param request - the request object containing the parameters to be provided in the API request
```

Comment:
another one here, looks like all the `list` methods are optional param.

Reply:
[05dab29](https://github.com/immutable/imx-core-sdk/pull/185/commits/05dab29a61dda6416ec0080c8c45a19130e36a68)